akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ReplicatedShardingExtensionImpl.scala

```diff
@@ -43,13 +43,14 @@ private[akka] final class ReplicatedShardingExtensionImpl(system: ActorSystem[_]
       thisReplica: Option[ReplicaId],
       settings: ReplicatedEntityProvider[M]): ReplicatedSharding[M] = {
     require(settings.replicas.nonEmpty, "Replicas must not be empty")
-    val typeName = settings.replicas.head._1.entity.typeKey.name
+    // typeName without the replica id
+    val typeName = settings.replicas.head._2
     val sharding = ClusterSharding(system)
     val initializedReplicas = settings.replicas.map {
       case (replicaSettings, typeName) =>
         // start up a sharding instance per replica id
         logger.infoN(
-          "Starting Replicated Event Sourcing sharding for replica [{}] (ShardType: [{}])",
+          "Starting Replicated Event Sourcing sharding for replica [{}] (ShardType: [{}], typeName [{}])",
```

Comment:
`typeName` is only being used inside `if (settings.directReplication)` further on. We can move it to inside that `if` block. 

It's also being shadowed by `case (replicaSettings, typeName)` here.

Reply:
also, `replicaToRegionOrProxy` is only used inside the `if`. Can also move.

Reply:
also, replicaToRegionOrProxy is only used inside the if. Can also move.

once `typeName` moves inside the `if` we can call it `typeNameWihtoutReplicaId`. To make it clear why were are giving preference to this one instead of the one from `initializedReplicas` that are explicitly ignoring.

Reply:
Good point.